### PR TITLE
proxy: update fluentd proxy to validate against 2.0 spec.

### PR DESCRIPTION
**Pull request by JDarDagran**
Add unit tests to CI.

### Problem

Thanks to @sekikn in https://github.com/OpenLineage/OpenLineage/pull/2208 a bug in current fluentd proxy was found. It needs update to validate against 2.0.x OpenLineage spec.

Closes: #2210 

### Solution

Change how facets are added to main spec file. Enrich error message for `oneOf` validation.

#### One-line summary:

Update fluentd proxy to validate against 2.0 spec.

### Checklist

- [x] You've [signed-off](https://github.com/OpenLineage/OpenLineage/blob/main/why-the-dco.md) your work
- [x] Your pull request title follows our [guidelines](https://github.com/OpenLineage/OpenLineage/blob/main/CONTRIBUTING.md#creating-pull-requests)
- [x] Your changes are accompanied by tests (_if relevant_)
- [x] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [x] Your comment includes a one-liner for the changelog about the specific purpose of the change (_if necessary_)
- [ ] You've versioned the core OpenLineage model or facets according to [SchemaVer](https://docs.snowplowanalytics.com/docs/pipeline-components-and-applications/iglu/common-architecture/schemaver) (_if relevant_)
- [x] You've added a [header](https://github.com/OpenLineage/OpenLineage/tree/main/.github/header_templates.md) to source files (_if relevant_)

----
SPDX-License-Identifier: Apache-2.0\
Copyright 2018-2023 contributors to the OpenLineage project

### CODE REVIEW

File: proxy/fluentd/lib/fluent/plugin/parser_openlineage.rb [hunk 1]
```diff
@@ -79,127 +100,83 @@ def rewrite_schema_to_include_facets(schema)
...
+                end
+              }
+              parents.each {|parent|
+              add_ref_as_parent_property(schema_json, parent, facet_name, property)
+              }
+            }
+            # include facets' definitions within schema
+            schema_json["$defs"] = schema_json["$defs"].merge(facet_schema["$defs"])
+        }
+        File.write('./test.json', JSON.dump(schema_json))
```
Comment (pawel-big-lebowski) [inline proxy/fluentd/lib/fluent/plugin/parser_openlineage.rb:128]: some debug left?

File: proxy/fluentd/lib/fluent/plugin/parser_openlineage.rb [hunk 2]
```diff
@@ -79,127 +100,83 @@ def rewrite_schema_to_include_facets(schema)
...
-              job_facets["properties"][property_name] = {  "$ref" => "#/$defs/" + facet_name }
-          }
-        else
-          job_facets["additionalProperties"] = { "type" => "object" }
+        
+        # list all the facets
+        Dir.glob("#{facets_path}/*.json").each { |facet_file|
+            facet_schema =  JSON.parse(
+              File.read(facet_file).gsub(
+                /"https:\/\/openlineage\.io\/spec\/\d-\d-\d\/OpenLineage\.json#\/\$defs\/([a-zA-Z]+)"/,
```
Comment (pawel-big-lebowski) [inline proxy/fluentd/lib/fluent/plugin/parser_openlineage.rb:108]: Juicy 🥝 🍓 🍌 🍏

**Review by pawel-big-lebowski — APPROVED** [review]
That's is something we were looking for. 
Minor comment to be verified. 
A deserved approval.

File: proxy/fluentd/lib/fluent/plugin/parser_openlineage.rb (hunk 1 above)
Comment (JDarDagran) [reply proxy/fluentd/lib/fluent/plugin/parser_openlineage.rb:128]: You're right, sorry!

**Pull request merged**